api/provider/aws/aws_test/load_balancer_list_test.go

```diff
@@ -0,0 +1,116 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadBalancerList(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "lb_id1",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name1",
+		},
+		{
+			EntityID:   "lb_id1",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id1",
+		},
+		{
+			EntityID:   "env_id1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name1",
+		},
+		{
+			EntityID:   "lb_id2",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name2",
+		},
+		{
+			EntityID:   "lb_id2",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "env_id2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	describeLoadBalancerPagesFN := func(input *elb.DescribeLoadBalancersInput, fn func(output *elb.DescribeLoadBalancersOutput, lastPage bool) bool) error {
```

Comment:
Do we expect only loadbalancers for a particular instance get returned from the api call? If not, we should throw in loadbalancers that don't belong to the layer0 instance to ensure they get filtered out.

Reply:
edit: just checked, the api call can return anything, so you should definitely add those extras